Improve the Readme (#1898)
official-stockfish/Stockfish: Readme.md

## Overview

[![Build Status](https://travis-ci.org/official-stockfish/Stockfish.svg?branch=master)](https://travis-ci.org/official-stockfish/Stockfish)
[![Build Status](https://ci.appveyor.com/api/projects/status/github/official-stockfish/Stockfish?svg=true)](https://ci.appveyor.com/project/mcostalba/stockfish)

[Stockfish](https://stockfishchess.org) is a free, powerful UCI chess engine derived from Glaurung 2.1. It is
not a complete chess program and requires a UCI-compatible GUI
(e.g. XBoard with PolyGlot, Scid, Cute Chess, eboard, Arena, Sigma Chess, Shredder, Chess
Partner or Fritz) in order to be used comfortably. Read the
documentation for your GUI of choice for information about how to use
Stockfish with it.

## Files

This distribution of Stockfish consists of the following files:

  * Readme.md, the file you are currently reading.

  * Copying.txt, a text file containing the GNU General Public License version 3.

  * src, a subdirectory containing the full source code, including a Makefile
    that can be used to compile Stockfish on Unix-like systems.

## UCI parameters

> Currently Stockfish exposes the following UCI options:

Reviewer: ```suggestion
Currently, Stockfish has the following UCI options:
```

  * #### Debug Log File

>     Write all communication from the engine and to the engine into a text file.

Reviewer: ```suggestion
    Write all communication to and from the engine into a text file.
```

  * #### Contempt
    A positive value for contempt favors middle game positions and avoids draws.

  * #### Analysis Contempt

>     By default, contempt is set to prefer the side to move. Set this option to White or Black to analyse with contempt for that side or set this option to Off to disable contempt.

Reviewer: ```suggestion
    By default, contempt is set to prefer the side to move. Set this option to "White" or "Black" to analyse with contempt for that side, or "Off" to disable contempt.
```

  * #### Threads

>     The number of CPU threads used for searching a position. Should equal the number of CPU cores for best performance.

Reviewer: ```suggestion
    The number of CPU threads used for searching a position. For best performance, set this equal to the number of CPU cores available.
```

  * #### Hash
    The size of the hash table in MB.

  * #### Clear Hash
    Clear the hash table.

  * #### Ponder
    Let Stockfish ponder its next move while the opponent is thinking.

  * #### MultiPV
    Output the N best lines (principle variantions, PVs) when searching. Leave at 1 for best performance.

  * #### Skill Level

>     Lower the Skill Level in order make Stockfish play weaker.

Reviewer 2: Missing word

  * #### Move Overhead
    Assume a time delay of x ms due to network and GUI overheads. This is useful to avoid losses on time in those cases.

  * #### Minimum Thinking Time
    Search for at least x ms per move. 

  * #### Slow Mover

>     Lower value will make Stockfish take less times in games, higher values will make it think longer.

Reviewer: ```suggestion
    Lower values will make Stockfish take less time in games, higher values will make it think longer.
```

  * #### nodestime
    Tells the engine to use nodes searched instead of wall time to account for elapsed time. Useful for engine testing.

  * #### UCI_Chess960
    An option handled by your GUI. If true, Stockfish will play Chess960.

  * #### UCI_AnalyseMode
    An option handled by your GUI.

  * #### SyzygyPath
    Path to the folders/directories storing the Syzygy tablebase files. Multiple directories are to be separated by ";" on Windows and by ":" on Unix-based operating systems. Do not use spaces around the ";" or ":".
    
    Example: `C:\tablebases\wdl345;C:\tablebases\wdl6;D:\tablebases\dtz345;D:\tablebases\dtz6`
    
    It is recommended to store .rtbw files on an SSD. There is no loss in storing the .rtbz files on a regular HD.

  * #### SyzygyProbeDepth

>     Minimum remaining search depth for which a position is probed. Set this option to a higher value to probe less agressively if you experience too much

Reviewer: ```suggestion
    Minimum remaining search depth for which a position is probed. Set this option to a higher value to probe less aggressively if you experience too much slowdown (in terms of nps) due to TB probing.
```

>     slowdown (in terms of nps) due to TB probing.

Reviewer: ```suggestion
```

Stockfish with it.

## Files

This distribution of Stockfish consists of the following files:

  * Readme.md, the file you are currently reading.

  * Copying.txt, a text file containing the GNU General Public License version 3.

  * src, a subdirectory containing the full source code, including a Makefile
    that can be used to compile Stockfish on Unix-like systems.

## UCI parameters

Currently, Stockfish has the following UCI options:

  * #### Debug Log File
    Write all communication to and from the engine into a text file.

  * #### Contempt
    A positive value for contempt favors middle game positions and avoids draws.

  * #### Analysis Contempt
    By default, contempt is set to prefer the side to move. Set this option to "White" or "Black" to analyse with contempt for that side, or "Off" to disable contempt.

  * #### Threads
    The number of CPU threads used for searching a position. For best performance, set this equal to the number of CPU cores available.

  * #### Hash
    The size of the hash table in MB.

  * #### Clear Hash
    Clear the hash table.

  * #### Ponder
    Let Stockfish ponder its next move while the opponent is thinking.

  * #### MultiPV

>     Output the N best lines (principle variantions, PVs) when searching. Leave at 1 for best performance.

Reviewer 3: `variations` instead of `variantions`

    Let Stockfish ponder its next move while the opponent is thinking.

  * #### MultiPV

>     Output the N best lines (principle variations, PVs) when searching. Leave at 1 for best performance.

Reviewer 2: Should be principal variations, not principle variations

set/unset some switches in the compiler command line; see file *types.h*
for a quick reference.

> ## Resource For Understanding the Code Base and participating in the project

Reviewer 3: Really weird title casing here. Headers for the other sections all just use normal sentence-casing, so should be the same here.

Author: Thanks for reading all this stuff. Not sure what you would like to read here, but I made a small modification anyway.

> Stockfish's improvement of the last couple of years has been a great community effort. There are a few ways to help contribute to its growth.

Reviewer 3: "over the last couple of years" is more idiomatic.

set/unset some switches in the compiler command line; see file *types.h*
for a quick reference.

## Resource For Understanding the Code Base and participating in the project

> Stockfish's improvement of the last couple of years have been a great community effort. You can help getting stronger in different ways.

Reviewer: ```suggestion
Stockfish's improvement of the last couple of years has been a great community effort. There are a few ways to help contribute to its growth.
```

### Donating hardware

> Improving Stockfish requires a massive amount of testing. You can simply donate your hardware by installing the [Fishtest Worker](https://github.com/glinscott/fishtest/wiki/Running-the-worker) and view the current tests on [Fishtest](http://tests.stockfishchess.org/tests).

Reviewer: ```suggestion
Improving Stockfish requires a massive amount of testing. You can donate your hardware resources by installing the [Fishtest Worker](https://github.com/glinscott/fishtest/wiki/Running-the-worker) and view the current tests on [Fishtest](http://tests.stockfishchess.org/tests).
```

### Improving the code

> If you want to help to improve the code, there are several valuable ressources to read:

Reviewer: ```suggestion
If you want to help improve the code, there are several valuable resources:
```

* [In this wiki,](https://www.chessprogramming.org) many techniques used in Stockfish are explained with a lot of background information.

> * [The section on Stockfish](https://www.chessprogramming.org/Stockfish) describes many features and techniques used by Stockfish. However, it is generic rather than being focused on Stockfish's precise implementation. Nethertheless, a helpful resource.

Reviewer 2: Typo, should be "Nevertheless"

> * The latest source can always be found on  [GitHub](https://github.com/official-stockfish/Stockfish). Discussions about Stockfish take place in the [FishCooking](https://groups.google.com/forum/#!forum/fishcooking) group and engine testing is done on [Fishtest](http://tests.stockfishchess.org/tests). If you want to help improve Stockfish, please read this [guideline](https://github.com/glinscott/fishtest/wiki/Creating-my-first-test) first, where the fundamentals about Stockfish development are explained.

Reviewer: ```suggestion
* The latest source can always be found on  [GitHub](https://github.com/official-stockfish/Stockfish). Discussions about Stockfish take place in the [FishCooking](https://groups.google.com/forum/#!forum/fishcooking) group and engine testing is done on [Fishtest](http://tests.stockfishchess.org/tests). If you want to help improve Stockfish, please read this [guideline](https://github.com/glinscott/fishtest/wiki/Creating-my-first-test) first, where the basics of Stockfish development are explained.
```

Author: thanks for your suggestions. i implemented them
